### DISCUSSION TITLE
How to run RTLAMR2MQTT and RTL_433 Home Assistant add-ons together on one RPi 4 with two RTL-SDR sticks?

### DISCUSSION THREAD

**Discussion by WH3080** [allangood/rtlamr2mqtt#110] [Q&A]
Hi, I apologize in advance for this perhaps stupid newbie question.
I successfully installed this setup and receive some old thermometers via a generic RTL-SDR stick (https://linux-hardware.org/index.php?id=usb:0bda-2838) on  Bus 001 Device 004: ID 0bda:2838 followling these instructions: https://community.home-assistant.io/t/home-assistant-add-on-rtl-433-with-mqtt-auto-discovery/260665 on a Raspberry Pi 4.
Now I added a second generic RTL-SDR stick (https://linux-hardware.org/index.php?id=usb:0bda-2838) on Bus 001 Device 006: ID 0bda:2838 and would like to run rtlamr2mqtt in parallel to the well working rtl_433. Two of my meters are smart: water http://fcc.io/EWQ100W Itron 100WRD Water ERT Type 11 903/927 Mhz and gas http://fcc.io/100GDLAN Itron 100GDLAN Gas ERT Type 12 908/926.8 Mhz and they are also listed here: https://github.com/bemasher/rtlamr. 
So far, I am unable to figure out how to "tell" HA, to continue to use this RTL-SDR Bus 001 Device 004: ID 0bda:2838 for the old thermometers etc. via rtl_433 (https://github.com/pbkhrv/rtl_433-hass-addons) and Bus 001 Device 006: ID 0bda:2838 for gas and water via https://github.com/allangood/rtlamr2mqtt. 
In the RTLAMR2MQTT configuration file I entered:
general:
  usb_reset: '006:001'
I was hoping this would select the 2nd RTL-SDR.
The system log shows this: 22-04-09 11:48:48 WARNING (MainThread) [supervisor.misc.tasks] Watchdog found a problem with 6713e36e_rtlamr2mqtt!
22-04-09 11:48:48 INFO (SyncWorker_6) [supervisor.docker.interface] Cleaning addon_6713e36e_rtlamr2mqtt application
22-04-09 11:48:50 INFO (SyncWorker_6) [supervisor.docker.addon] Starting Docker add-on 6713e36e/aarch64-addon-rtlamr2mqtt with version 1.7.0
The message keeps repeating itself endlessly. Uninstalling and rebuilding have not helped. The reason I plugged in two RTL-SDR sticks was that I was hoping that I can receive two frequency ranges in parallel with two independent hardware devices, since I thought trying it with one stikc would cause conflicts of rtl_433 and rtlamr. But maybe that is wrong?
And, when I type in "rtl_tcp" or "rtlamr" into the SSH sheel inside HA, I don't see anything but an error message. 
[core-ssh /]$ rtl_tcp
bash: rtl_tcp: command not found
[core-ssh /]$ rtlamr
bash: rtlamr: command not found
Also, identifying the meters via this command "docker run --rm -ti -e LISTEN_ONLY=yes -e RTL_MSGTYPE="all" --device=/dev/bus/usb:/dev/bus/usb allangood/rtlamr2mqtt" probably requires some adjustments I guess?
I have found some numbers on  the smart meters, but to be sure, I would like to use this feature to "discover them".
Any help would be greatly appreciated!

**Comment by allangood** [comment]
RTL_TCP has the parameter `-d` to define which device to use, i.e.: `-d 0` for the first device, `-d 1` for the second and so on.

With this rtlamr2mqtt application, you can change the custom_parameters section to add this parameter, like this:
```
custom_parameters:
  rtltcp: "-s 2048000 -d 1"
  rtlamr: "-unique=true -symbollength=32"
```
Refer to the other container documentation to learn how to add the parameter there.

Regards.

**Reply by WH3080** [reply]
Thank you Allan. Will try. Greetings from Alberta ;-)

**Reply by 42Network** [reply]
Hi. I'd like to ask a followup question. I have the same situation, wherein I have two USB RTL-SDR dongles connected to my HA system. One has an antenna kinda outside for detecting water meter readings which will be associated with RTLAMR. The other dongle is connected to an antenna towards the center of my house to pick up 433MHz transmissions from thermometers, etc.

Can you elaborate on how the Device ID is assigned? Will one dongle CONSISTENTLY be assigned ID 0 and the other 1? How does the underlying OS make the assignment? I can see from dmesg and lsusb that when I unplug and replug my dongles, they get different and increasing Device numbers, but I am not sure these Device numbers align with the RTL-SDR drivers idea of device number. 

E.g. my two dongles are USB device number 16 and 17 here, but earlier in the day they were 11 and 12
```
[1390258.003088] usb 1-1: USB disconnect, device number 15
[1390284.196501] usb 1-1: new high-speed USB device number 16 using ehci-pci
[1390284.335801] usb 1-1: New USB device found, idVendor=0bda, idProduct=2838, bcdDevice= 1.00
[1390284.335808] usb 1-1: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[1390284.335810] usb 1-1: Product: RTL2838UHIDIR
[1390284.335811] usb 1-1: Manufacturer: Realtek
[1390284.335813] usb 1-1: SerialNumber: 00000102
[1394385.018646] usb 1-2: new high-speed USB device number 17 using ehci-pci
[1394385.156452] usb 1-2: New USB device found, idVendor=0bda, idProduct=2838, bcdDevice= 1.00
[1394385.156468] usb 1-2: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[1394385.156470] usb 1-2: Product: RTL2838UHIDIR
[1394385.156472] usb 1-2: Manufacturer: Realtek
[1394385.156473] usb 1-2: SerialNumber: 00000101
```

**Reply by cybergrimes** [reply]
> Hi. I'd like to ask a followup question. I have the same situation, wherein I have two USB RTL-SDR dongles connected to my HA system. 
> ```

@42Network 
Did you ever sort this or have a 2 adapter system? I'm running RTLAMR now and looking to expand to a RTL_433 for sensors like you have but unsure if I'll be able to get them working at the same time reliably as you've described here. Currently using the "device_id" configuration parameter for RTLAMR but it looks like the RTL_433 add-on doesn't let you specify this way.
